packages/SwingSet/misc-tools/ingest-xs-snapshot.js

```diff
@@ -0,0 +1,744 @@
+/* eslint-disable */
+// import '@endo/init';
+// import { E } from '@endo/far';
+import fs from 'fs';
+import zlib from 'zlib';
+import process from 'process';
+import sqlite3 from 'better-sqlite3';
+
+// super-experimental tool to parse an XS heap snapshot file (as
+// extracted from kerneldb by extract-snapshot.js) and populate a
+// sqlite DB with its contents for later perusal. I have probably
+// misunderstood a lot about the format, and there are definitely some
+// fencepost errors that corrupt the relationships between slots and
+// the things they reference. Run this like:
+//
+// node misc-tools/ingest-xs-snapshot.js v6-2-b2ed...xss output.sqlite
+// sqlite3 output.sqlite .schema
+
+function makeDB(dbfn) {
+  const db = sqlite3(dbfn);
+  db.exec(`PRAGMA journal_mode=WAL`);
+  // these tables hold the snapshot contents more-or-less verbatim
+  db.exec(`CREATE TABLE chunk (chunk BLOB)`);
```

Comment:
Depending on your intended mode of use (i.e., whether or not a one of these databases is intended for the analysis of just a single snapshot or more than one), I suspect you either want to create the tables with `CREATE TABLE IF NOT EXISTS` or on startup make sure the database file doesn't exist (or delete it if it does).

Reply:
yeah, this is definitely a one-shot thing.. if I add more data, I'd just re-ingest the original snapshot